crates/context_aware_config/src/api/default_config/handlers.rs

```diff
@@ -184,3 +194,59 @@ async fn get(db_conn: DbConnection) -> superposition::Result<Json<Vec<DefaultCon
     let result: Vec<DefaultConfig> = default_configs.get_results(&mut conn)?;
     Ok(Json(result))
 }
+
+pub fn check_key_in_context_overrides(
+    key: &String,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+) -> superposition::Result<Vec<String>> {
+    let result: Vec<Context> = contexts.load(conn).map_err(|err| {
+        log::error!("failed to fetch contexts with error: {}", err);
+        db_error!(err)
+    })?;
+
+    let mut context_ids = vec![];
+    for context in result.iter() {
+        from_value::<Map<String, Value>>(context.override_.to_owned())
+            .map_err(|err| {
+                log::error!("failed decode override into object: {}", err);
+                unexpected_error!(err)
+            })?
+            .get(key)
+            .map_or((), |_| context_ids.push(context.id.to_owned()))
+    }
+    Ok(context_ids)
+}
+
+#[delete("/{key}")]
+async fn delete_default_config(
+    path: Path<String>,
+    db_conn: DbConnection,
+    user: User,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+
+    let key = path.into_inner();
+    fetch_default_key(&key, &mut conn)?;
+    let context_ids = check_key_in_context_overrides(&key, &mut conn)?;
```

Comment:
`check_key_in_context_overrides` is an internal function, any error returned by should not be given back in response, rather a simple message as `Something went wrong` should be good enough.
we can use `map_err` here to achieve the same.

Reply:
@ShubhranshuSanjeev the point is to inform the user about the key being used in the following contexts.

Reply:
@ShubhranshuSanjeev this function returns two types of error db_error and unexpected error, 
both types were already handled in macro, message were passed according to that only